python/cugraph_service/cugraph_service_server/cugraph_handler.py

```diff
@@ -669,13 +717,53 @@ def get_graph_edge_data(
             columns = property_keys
         if types == []:
             types = None
+        if isinstance(G, (PropertyGraph, MGPropertyGraph)):
+            try:
+                df = G.get_edge_data(edge_ids=ids, columns=columns, types=types)
+            except KeyError:
+                df = None
+        else:
+            if columns is not None:
+                raise CugraphServiceError(
+                    f"Graph does not contain properties. {columns}"
+                )
 
-        try:
-            df = pG.get_edge_data(edge_ids=ids, columns=columns, types=types)
-            if isinstance(df, dask_cudf.DataFrame):
-                df = df.compute()
-        except KeyError:
-            df = None
+            # Get the edgelist; API expects edge id, src, dst, type
+            df = G.edgelist.edgelist_df
+
+            if G.edgeIdCol in df.columns:
+                if ids is not None:
+                    ids = cudf.Series(ids)
+                    if self.is_mg:
+                        ids = dask_cudf.from_cudf(ids, npartitions=self.num_gpus)
+                    df = df.reindex(df[G.edgeIdCol]).loc[ids]
+            else:
+                if ids is not None:
+                    raise CugraphServiceError("Graph does not have edge ids")
+                df[G.edgeIdCol] = df.index
+
+            if G.edgeTypeCol in df.columns:
+                if types is not None:
+                    df = df[df[G.edgeTypeCol].isin(types)]
+            else:
+                if types is not None:
+                    raise CugraphServiceError("Graph does not have typed edges")
+                df[G.edgeTypeCol] = ""
+
+            src_col_name = (
+                G.renumber_map.renumbered_src_col_name if self.is_mg else "src"
+            )
+            dst_col_name = (
+                G.renumber_map.renumbered_dst_col_name if self.is_mg else "dst"
+            )
+            if G.is_renumbered():
+                df = G.unrenumber(df, src_col_name, preserve_order=True)
+                df = G.unrenumber(df, dst_col_name, preserve_order=True)
```

Comment:
We should probably look into doing these calls at once . Also, i think they dont work with `dask_cudf` frames . CC: @jnke2016  as he was looking into something similar.

Reply:
The number map should accept both cudf and dask_cudf dataframes.

Reply:
Right, this shouldn't be an issue if it is a dataframe being `unrenumbered`. The issue with Random_walk was unique as I was `unrenumbering` a `dask_cudf.Series`(vertex_paths) which had to match the order of another `dask_cudf.Series`(edge_paths) of different size.